### ISSUE TITLE
[FEAT]:  Supporting gradient accumulation 

### ISSUE THREAD

**Issue by msalhab96** [msalhab96/SpeeQ#22]
### Description

Adding gradient accumulation across training steps in the trainers
